src/components/Divider/src/Divider.stories.tsx

```diff
@@ -18,98 +19,58 @@ export default {
   component: Divider,
 } as Meta;
 
-const Template: Story<DividerProps> = (args: DividerProps) => (
+const Template: Story<DividerProps> = (args: DividerProps) => {
+  return args.orientation === "vertical" ? (
+    <Grid container>
+      <p>I am to the left of the divider.</p>
+      <Divider {...args} />
+      <p>I am to the right of the divider.</p>
+    </Grid>
+  ) : (
+    <List>
+      <ListItem button>
+        <ListItemIcon>
+          <InboxIcon />
+        </ListItemIcon>
+        <ListItemText primary="I am above the divider." />
+      </ListItem>
+      <Divider {...args} />
```

Comment:
I know the `<Divider/>` is used for this purpose in Material-UI, but is this really something to strive for? Couldn't this be implemented as `list--dividers` or something like that with `border-bottom`? I feel that ideally the Divider component has `role="separator"` by default, but that has a certain meaning and effect for screen readers. For list items that would not be appropriate, whereas for separating texts or separating toolbar buttons into groups it would be appropriate.

Reply:
Perhaps it's a better idea to split them into different components as you mentioned earlier. Our use case for the divider is to split visual elements, not necessarily a screen reader separator.

Reply:
If this Divider element is for the presentational variant, I don't think it should be implemented with an `<hr>` like in Material-UI, because that has `role="separator"` implied. Also their suggestion to use the inset variant that renders as `<li role="separator">` doesn't seem very accessible. It feels to me that it would only be appropriate to separate groups of list items, not to separate every list item.

Also the story of separating text content like `<p>...</p><Divider/><p>...</p>` should not be included in the examples for a visual divider, since those should have the "semantic" separator.

Anyway, I agree there should be two separate components in that case. I think you would need to implement the visual divider yourself and not rely on Material-UI in this case.

Reply:
The vertical variant separates two `<p>` elements in a MUI `<Grid>` component, separating text. Whilst this Divider component is not the separator component you are referring to, I do think that it does also serve that purpose. In the background this component is just an hr, so I think applying role="separator" to this Divider may result in the same accessibility-wise.

Reply:
> Also the story of separating text content like `<p>...</p><Divider/><p>...</p>` should not be included in the examples for a visual divider, since those should have the "semantic" separator.

The purpose of this example wasn't so much to split text, it was moreso to demo the ability to vertically divide two components. Will push a commit with different components in a bit.